e2e/test/MsTestMethod.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    public class MsTestMethod : TestMethodAttribute
+    {
+        private static TestLogging _logger = TestLogging.GetInstance();
+
+        public override TestResult[] Execute(ITestMethod testMethod)
```

Comment:
The test results are not exposed anywhere so the way to get that to logs is by overriding the Execute method and logging the value directly from here on failure.